homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,280 @@
+        native_unit_of_measurement="gpm",
+        suggested_display_precision=1,
+        value_fn=lambda device: device.drop_api.current_flow_rate(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=PEAK_FLOW_RATE,
+        translation_key=PEAK_FLOW_RATE,
+        icon="mdi:shower-head",
+        native_unit_of_measurement="gpm",
+        suggested_display_precision=1,
+        value_fn=lambda device: device.drop_api.peak_flow_rate(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=WATER_USED_TODAY,
+        translation_key=WATER_USED_TODAY,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=1,
+        value_fn=lambda device: device.drop_api.water_used_today(),
+        state_class=SensorStateClass.TOTAL,
+    ),
+    DROPSensorEntityDescription(
+        key=AVERAGE_WATER_USED,
+        translation_key=AVERAGE_WATER_USED,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.average_water_used(),
+        state_class=SensorStateClass.TOTAL,
+    ),
+    DROPSensorEntityDescription(
+        key=CAPACITY_REMAINING,
+        translation_key=CAPACITY_REMAINING,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.capacity_remaining(),
+        state_class=SensorStateClass.TOTAL,
+    ),
+    DROPSensorEntityDescription(
+        key=CURRENT_SYSTEM_PRESSURE,
+        translation_key=CURRENT_SYSTEM_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.PSI,
+        suggested_display_precision=1,
+        value_fn=lambda device: device.drop_api.current_system_pressure(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=HIGH_SYSTEM_PRESSURE,
+        translation_key=HIGH_SYSTEM_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.PSI,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.high_system_pressure(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=LOW_SYSTEM_PRESSURE,
+        translation_key=LOW_SYSTEM_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=UnitOfPressure.PSI,
+        suggested_display_precision=0,
+        value_fn=lambda device: device.drop_api.low_system_pressure(),
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    DROPSensorEntityDescription(
+        key=BATTERY,
+        device_class=SensorDeviceClass.BATTERY,
```

Comment:
If it's a battery measurement for the battery that powers the device we normally set the `entity_category` attribute to `EntityCategory.DIAGNOSTIC`. If the battery is the main feature, like on a UPS or power bank, it's not diagnostic.

https://developers.home-assistant.io/docs/core/entity#registry-properties

Reply:
```suggestion
        device_class=SensorDeviceClass.BATTERY,
        entity_category=EntityCategory.DIAGNOSTIC,
```

Reply:
The entity category was added in 3766113934397bf3d508ec3ddb37f27c6f2d5868.  The battery for most of our devices is a backup power source to allow operation during a power outage, so I'd agree it's of a diagnostic nature.